informational/req-res-reliability.md

```diff
@@ -0,0 +1,62 @@
+---
+title: REQ-RES-RELIABILITY
+name: Request-response protocols reliability
+category: Best Current Practice
+tags: [informational]
+editor: Oleksandr Kozlov <oleksandr@status.im>
+contributors:
+- Prem Chaitanya Prathi <prem@status.im>
+- Danish Arora <danish@status.im>
+---
+
+## Abstract
+This RFC describes set of instructions used across different [WAKU2](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/10/waku2.md?plain=1#L3) implementations for improved reliability in request-response protocols such as [WAKU2-LIGHTPUSH](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/19/lightpush.md?plain=1#L3C11-L3C26) and [WAKU2-FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md?plain=1#L3).
+
+## Motivation
+
+Descriptions of mentioned protocols do not define some of the real world use cases that are oftenly observed in unreliable network environment. Such use cases can be: recovery from offline state, decrease rate of missed messages, increase probability of messages being broadcasted within the network.
+
+## Suggestions
+
+### Node health
+
+As a useful metric to define and implement for determining quality of provided service by a node:
+- unhealthy - no peer connections are available regardless of protocol;
+- minimally healthy:
+  - Relay has less than 4 peers connected;
+  - Filter and LightPush has one per each peer connection available;
+- sufficiently healthy:
+  - Relay has minimum 4 peers connected;
+  - more than 1 connection in Filter and at least 2 connections available in LightPush;
+
+### Peers and connection management
+
+- Each protocols should retain a pool of reliable peers. In case a protocol failed to use any peer more than once - connection to it should be dropped and new peer should be added to the pool instead. 
```

Comment:
addressed here - https://github.com/waku-org/specs/pull/18/commits/594a1ecf13ab5665e749b9adaab168fa77020a25

> We can leave precise definition of "bad service" to the individual protocol specs.

do you see it being a follow up to this PR or, perhaps, I can add it as separate section that defines more precisely `fails to serve protocol request` (as I defined it rn)

> We can then continue to say that we recommend a very strict approach of removing service nodes after even 1 failure. (Do we really recommend this, though? If we fail to connect to a service node, should we completely remove it from the pool even if it might just be a temporary connection failure?)

I think this change I did before we made actual code change so that's why it was `more than once` and I am changing it to `fails to serve protocol request from a light node 3 times`. I am not sure if it is best framing though